docs/self-managed/concepts/pvc-resize-guide.md

```diff
@@ -0,0 +1,163 @@
+```bash
+kubectl get storageclass -o yaml | grep allowVolumeExpansion
+```
+
+Or for a specific class:
+
+```bash
+kubectl get storageclass <your-storage-class> -o yaml | grep -A5 -B5 allowVolumeExpansion
+```
+
+### Step 1: Check Current PVCs
+```bash
+kubectl get pvc -l app.kubernetes.io/component=data -o wide
+```
+
+### Step 2: Resize PVCs Directly
+```bash
+# Example: Resize two Elasticsearch data PVCs
+kubectl patch pvc data-elasticsearch-data-0 -p '{"spec":{"resources":{"requests":{"storage":"200Gi"}}}}'
+kubectl patch pvc data-elasticsearch-data-1 -p '{"spec":{"resources":{"requests":{"storage":"200Gi"}}}}'
+```
+
+
+#### Monitor:
+
+```bash
+kubectl get pvc -l app.kubernetes.io/component=data -w
+```
+
+### Step 3 (Optional): Update Helm Values
+```yaml
+# values.yaml for bitnami ES chart
+data:
+    persistence:
+        size: 200Gi
+```
+
+```bash
+helm upgrade elasticsearch oci://registry-1.docker.io/bitnamicharts/elasticsearch \
+--version 21.6.3 \
+-f values.yaml
+```
+
+### Step 4: Verify Expansion
+```bash
+kubectl exec -it elasticsearch-data-0 -- df -h /bitnami/elasticsearch/data
+kubectl exec -it elasticsearch-data-1 -- df -h /bitnami/elasticsearch/data
+
+kubectl get pvc -l app.kubernetes.io/component=data
+```
+
+
+### ✅ Summary (If Expansion Is Supported)
+
+No downtime required
+
+One-way operation (can’t shrink)
+
+Supported by EBS, GCE PD, Azure Disk, etc.
+
+## Approach 2: Resize PVC using CSI Snapshots
+❌ If Expansion Is NOT Supported — Use CSI Snapshot + Clone
+
+### Step 1: Check for CSI Snapshot Support
+```bash
+kubectl api-resources | grep volumesnapshot
+```
+
+Should return:
+
+```bash
+volumesnapshots                        snapshot.storage.k8s.io/v1
+volumesnapshotcontents                snapshot.storage.k8s.io/v1
+volumesnapshotclasses                 snapshot.storage.k8s.io/v1
+```
+
+Also check:
+
+```bash
+kubectl get volumesnapshotclass
+```
+
+## Step 2: Create a VolumeSnapshot
+
+```yaml
+# snapshot.yaml
+apiVersion: snapshot.storage.k8s.io/v1
+kind: VolumeSnapshot
+metadata:
+    name: data-elasticsearch-data-0-snap
+spec:
+    volumeSnapshotClassName: <your-snapshot-class>
+    source:
+        persistentVolumeClaimName: data-elasticsearch-data-0
+```
+
+```bash
+kubectl apply -f snapshot.yaml
+```
+
+Wait until snapshot is ready:
+
+```bash
+kubectl get volumesnapshot
+```
+
+Step 3: Create a New PVC from the Snapshot (Larger Size)
+
+```yaml
+# new-pvc.yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+    name: data-elasticsearch-data-0-expanded
+spec:
+    storageClassName: <your-expandable-storageclass>
+    dataSource:
+        name: data-elasticsearch-data-0-snap
+        kind: VolumeSnapshot
+        apiGroup: snapshot.storage.k8s.io
+    accessModes:
+      - ReadWriteOnce
+    resources:
+        requests:
+            storage: 200Gi
+```
+
+```bash
+kubectl apply -f new-pvc.yaml
+```
+
+## Step 4: Replace Original PVC (Optional Downtime)
```

Comment:
shouldn't the first step be that you take the system offline to allow for a coherent volume snapshot?
Downtime is required in any case.